lib/compilation_engine.rb

```diff
@@ -282,6 +279,8 @@ def compile_subroutine_body
       compile_var_dec
     end
 
+    @vm_writer.write_function("#{@class_name}.#{@subroutine_name}", @symbol_table.var_count(:VAR))
```

Comment:
Right, good! Unlike the [earlier situation](https://github.com/wildmaples/jack-compiler/pull/17#discussion_r785951200) with expression lists, this is safe because subroutine definitions can’t be nested, so we can be sure we’re getting the `@subroutine_name` we just set inside `#compile_subroutine`.